src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -288,6 +289,10 @@ export type OncokbCancerGene = {
 export class StudyViewPageStore {
     private reactionDisposers: IReactionDisposer[] = [];
 
+    private colorMap : Map<string, string>;
+    private chartSet : Set<string>;
```

Comment:
I do feel like we already have an object keeping track of which charts are active, by chart id rather than by clinical attribute id. so I dont think chartSet is necessary. right @zhx828 ?

Reply:
Yeah, but the logic here is that you just want to keep the color consistent. Why bother to check whether the chart is available? You could generate color counts using getClinicalDataCountWithColorByClinicalDataCount, then do a quick check whether the (attribtue id + attribute value) exists in the color cache, if it is then replace.